src/modules/update_project/app/update_project_controller.py

```diff
@@ -0,0 +1,62 @@
+from src.shared.helpers.errors.controller_errors import MissingParameters
+from src.shared.helpers.errors.domain_errors import EntityError
+from src.shared.helpers.errors.usecase_errors import NoItemsFound
+from src.shared.helpers.external_interfaces.external_interface import IRequest, IResponse
+from src.shared.helpers.external_interfaces.http_codes import OK, NotFound, BadRequest, InternalServerError
+from .update_project_viewmodel import UpdateProjectViewmodel
+from .update_project_usecase import UpdateProjectUsecase
+
+class UpdateProjectController:
+    def __init__(self, usecase: UpdateProjectUsecase) -> None:
+        self.UpdateProjectUsecase = usecase
+
+
+    def __call__(self, request: IRequest) -> IResponse:
+        try:
+            if not request.data.get('code'):
+                raise MissingParameters('code')
+            
+            if request.data.get('new_name') is None:
```

Comment:
que tipo de update é?

ele tem que receber tudo pra dar update? (hard)

ou ele pode receber só o que vai dar update?

se for um kvp (key value par), ele da update só no que recebe é melhor (acho que foi o que vc fez no mock)

então n faz sentido essa validação se for None levantar erro, né?